int32  actionType
int32  row
int32  column
int32  numChars
int32  attributes
string text
string comment
